content/en/apps/guides/integrations/best-practices/rapidpro.md

```diff
@@ -0,0 +1,108 @@
+- Intro and outro messages
+   - These serve the purpose of giving the survey details such as background of the survey, number of questions, data protection, whether there shall be follow up, SMS billing, thank you notes etc. Include these if possible.
+- Consent and opt out
+   - Include questions that give the respondent an opportunity to opt in or out of the survey. If they opt out, do not send a follow up text.
+- SMS length
+   - Keep it short, to the size of one SMS (160 characters). Longer messages may not display well on the recipients’ devices.
+
+### Flow programming
+- Utilize skip logic to ask relevant questions only e.g. number of children if only they have children.
+- Automate error messages to to validate responses e.g. You submitted an invalid response. Reply 1 for “Yes”, 2 for “No”. These can be customized.
+- Use reminders and follow-up messages selectively on respondents by first differentiating those that completed a flow versus those that did not. This reduces spamming.
+- Customize messages as much as possible by pre-populating information already known such as name, to target respondents, especially when phones are shared among clients.
+- Translate all messages, especially when deploying in a multi-lingual environment. This ensures that respondents fully understand the survey in their language.
+- Beware of the timing of the surveys that directly affects response rates. From experience, sending questions when respondents are busy with their errands during the day ultimately leads to low response rates as opposed to evenings when they are done for the day.
+- Make sure you handle unsolicited responses by redirecting such to for example, a flow that eventually alerts concerned individuals e.g. reports of an outbreak etc.
+
+### Configuration
+1. Persist a unique identifier for each respondent entering the flows. This identifier shall be used to link flow runs to the recipients to make data analysis smooth.
+2. Use consistent language and design patterns.
+3. Be mindful of when you save data back to the CHT. From experience, this should happen at major milestones in a flow e.g. end of a flow or before sending a payload to an API endpoint.
+4. Utilize `globals`, shared values that can be referenced in flows, as well as broadcasts and campaigns, within your account referenced by `@globals.value_name`. This avoids re-entry of values in various components.
+5. Beware of concurrency that is not supported in RapidPro. Concurrency refers to a situation whereby one contact participates in more than one flows at the same time. Whenever this happens, the former flow shall be interrupted in favor of the latter. This can result in respondents exiting flows before completion, which impacts data quality. A master flow that controls individual flows is recommended. 
```

Comment:
```suggestion
5. Beware of concurrency that is not supported in RapidPro. Concurrency refers to a situation whereby one contact participates in more than one flows at the same time. Whenever this happens, the former flow shall be interrupted in favor of the latter. This can result in respondents exiting flows before completion, which is a confusing user experience and results in poor survey data. A master flow that spins up individual flows may be useful to consider. 
```

i think this was good for goma but may not be a blanket recommendation

Reply:
I think it's good to have it since we faced the same issue in CIHA and Malawi